### PR TITLE
CSS support for custom components added

### PR DESCRIPTION
**Pull request by enchev**
Implemented part of https://github.com/NativeScript/NativeScript/issues/389 and https://github.com/NativeScript/NativeScript/issues/146

### CODE REVIEW

**Comment by enchev** [comment]
run ci

**Comment by enchev** [comment]
run ci

**Comment by enchev** [comment]
run ci

File: ui/builder/builder.ts [hunk 1]
```diff
@@ -208,6 +214,12 @@ function loadCustomComponent(componentPath: string, componentName?: string, attr
         result = componentBuilder.getComponentModule(componentName, componentPath, attributes, context);
     }
 
+    // Add component CSS file if exists.
+    var cssFileName = fileName.replace(".xml", ".css");
+    if (fs.File.exists(cssFileName) && parentPage) {
+        parentPage.addCssFile(cssFileName);
```
Comment (vakrilov) [inline ui/builder/builder.ts]: What about the case when the custom component is inside a template - will it load its CSS then?

File: ui/builder/builder.ts [hunk 2]
```diff
@@ -208,6 +214,12 @@ function loadCustomComponent(componentPath: string, componentName?: string, attr
         result = componentBuilder.getComponentModule(componentName, componentPath, attributes, context);
     }
 
+    // Add component CSS file if exists.
+    var cssFileName = fileName.replace(".xml", ".css");
```
Comment (vakrilov) [inline ui/builder/builder.ts]: Note: potential problem here: If we have `myComponent.xml`, but 2 CSS files: `myComponent.land.css` and `myComponent.port.css`. It is best to use the file-name-resolver module when figuring out which file to load

File: ui/builder/builder.ts [hunk 3]
```diff
@@ -218,7 +230,7 @@ export function load(arg: any): view.View {
     if (arguments.length === 1) {
         if (!types.isString(arguments[0])) {
             var options = <definition.LoadOptions>arguments[0];
-            componentModule = loadCustomComponent(options.path, options.name, undefined, options.exports);
+            componentModule = loadCustomComponent(options.path, options.name, undefined, options.exports, options.page);
```
Comment (hdeshev) [inline ui/builder/builder.ts]: Does it seem like a good idea to log/trace if we got no page here? Might be useful to people debugging "my component's CSS doesn't load" problems.

**Comment by hdeshev** [comment]
gaga

**Comment by nsndeck** [comment]
:+1:

File: ui/builder/builder.ts (hunk 3 above)
Comment (enchev) [inline ui/builder/builder.ts]: Done!

File: ui/builder/builder.ts (hunk 2 above)
Comment (enchev) [inline ui/builder/builder.ts]: file resolver added

**Comment by vakrilov** [comment]
:+1:

**Pull request merged**